components/contribution-flow/ContributionFlowSuccess.js

```diff
@@ -124,6 +132,41 @@ class ContributionFlowSuccess extends React.Component {
     data: PropTypes.object,
   };
 
+  componentDidUpdate() {
+    const {
+      router: { query: queryParams },
+      data: { order },
+    } = this.props;
+    if (order && queryParams.redirect) {
+      if (isValidExternalRedirect(queryParams.redirect)) {
+        const url = new URL(queryParams.redirect);
+        url.searchParams.set('orderId', order.legacyId);
+        url.searchParams.set('orderIdV2', order.id);
+        url.searchParams.set('status', order.status);
+        const transaction = find(order.transactions, { type: TransactionTypes.CREDIT });
```

Comment:
I understand this is duplicated code. If this valid here, do we have transactions available?